### DISCUSSION TITLE
Is it possible to set the visible Y-values?

### DISCUSSION THREAD

**Discussion by Tyler-Lopez** [patrykandpatrick/vico#847] [Questions] [answered]
### Question

Is it possible to set the visible Y-values? I notice that it is certainly the case that you can obviously view the values such that they fit the X-axis, but they don't seem to fit the Y-axis. For example, if my values are all between 100-200, the y-axis still starts at 0.

Is there anything I'm missing here?

### UI framework(s)

Jetpack Compose

**Comment by Tyler-Lopez** [comment]
I am answering myself again here with the solution as I understand it. Looking forward to more input if there is a more appropriate way. Obviously this could be remembered, but this is just an example of locking the y-axis between 60 and 120.

``` kotlin
            axisValueOverrider = object : AxisValueOverrider {
                    override fun getMinY(minY: Double, maxY: Double, extraStore: ExtraStore): Double {
                        return 60.0
                    }

                    override fun getMaxY(minY: Double, maxY: Double, extraStore: ExtraStore): Double {
                        return 120.0
                    }
                },
```

**Comment by Gowsky** [comment, marked as answer]
Hello, @Tyler-Lopez. `AxisValueOverrider` (`CartesianLayerRangeProvider` in newer versions) is the correct API. Here's how to achieve the result that you initially described:

```kt
object : AxisValueOverrider {
    override fun getMinY(minY: Double, maxY: Double, extraStore: ExtraStore) = minY

    override fun getMaxY(minY: Double, maxY: Double, extraStore: ExtraStore) = maxY
```

The `minY` and `maxY` arguments are the bounds of the data's _y_ range. As described [in the wiki](https://patrykandpatrick.com/vico/wiki/cartesian-charts/layers/#cartesianlayerrangeprovider), it's important that you obtain these values via these parameters.

As also mentioned in the wiki, when you need to lock the _y_ range for the entire lifecycle of the chart (as in your example code), you can use the `fixed` factory function.
